### PR TITLE
fix(core): introduce state when the modal has been initialized

### PR DESCRIPTION
**Pull request by Jammy-Louie**

### CODE REVIEW

**Comment by Jammy-Louie** [comment]
@christopherthielen just a proposal to fix the problem we ran into yesterday.

File: app/scripts/modules/core/src/modal/wizard/WizardModal.tsx [hunk 1]
```diff
@@ -62,19 +63,25 @@ export class WizardModal<T = {}> extends React.Component<IWizardModalProps<T>, I
...
+  public componentDidMount(): void {
+    this.setState({ initialized: true });
+  }
+
   public onWizardPageAdded = (wizardPage: WizardPage<T>): void => {
     this.setState(prevState => {
       const pages = prevState.pages.concat(wizardPage);
       const currentPage = prevState.currentPage || pages[0];
-      return { pages, currentPage };
+      const initialized = prevState.initialized;
```
Comment (christopherthielen) [inline app/scripts/modules/core/src/modal/wizard/WizardModal.tsx]: this shouldn't be necessary (`setState()` with the same value of `initialized`)

File: app/scripts/modules/core/src/modal/wizard/WizardModal.tsx [hunk 2]
```diff
@@ -62,19 +63,25 @@ export class WizardModal<T = {}> extends React.Component<IWizardModalProps<T>, I
...
+      return { pages, currentPage, initialized };
     }, this.revalidate);
   };
 
   public onWizardPageRemoved = (wizardPage: WizardPage<T>): void => {
     this.setState(prevState => {
       const pages = without(prevState.pages, wizardPage);
       const currentPage = prevState.currentPage || pages[0];
-      return { pages, currentPage };
+      const initialized = prevState.initialized;
```
Comment (christopherthielen) [inline app/scripts/modules/core/src/modal/wizard/WizardModal.tsx]: same

File: app/scripts/modules/core/src/modal/wizard/WizardModal.tsx [hunk 3]
```diff
@@ -171,7 +178,7 @@ export class WizardModal<T = {}> extends React.Component<IWizardModalProps<T>, I
               <Modal.Header>{heading && <h3>{heading}</h3>}</Modal.Header>
 
               <Modal.Body>
-                {loading ? (
+                {loading && initialized ? (
```
Comment (christopherthielen) [inline app/scripts/modules/core/src/modal/wizard/WizardModal.tsx]: should this be `loading || !initialized` ?  maybe it doesn't make any difference.

File: app/scripts/modules/core/src/modal/wizard/WizardModal.tsx (hunk 1 above)
Comment (Jammy-Louie) [reply app/scripts/modules/core/src/modal/wizard/WizardModal.tsx]: 👍

File: app/scripts/modules/core/src/modal/wizard/WizardModal.tsx (hunk 2 above)
Comment (Jammy-Louie) [reply app/scripts/modules/core/src/modal/wizard/WizardModal.tsx]: 👍

File: app/scripts/modules/core/src/modal/wizard/WizardModal.tsx (hunk 3 above)
Comment (Jammy-Louie) [reply app/scripts/modules/core/src/modal/wizard/WizardModal.tsx]: 👍

**Comment by Jammy-Louie** [comment]
changes made

**Pull request merged**